src/variable_info_detail.hpp

```diff
@@ -16,80 +16,150 @@
 
 #pragma once
 
-#include <string>
 #include "config.hpp"
+#include "utils/const_clone.hpp"
+
+#include <string>
+#include <type_traits>
 
-namespace variable_info_detail
+class invalid_variablename_exception : public std::exception
 {
-	enum variable_info_type {vit_const, vit_create_if_not_existent, vit_throw_if_not_existent,  };
-	enum variable_info_state_type {
-		state_start = 0, // for internal use
-		                 // only used at the 'starting_pos' of the variable_info::calculate_value algorithm
-		state_named,     // the result of .someval this can eigher man an attribute value or an
-		                 // child range
-		state_indexed,   // the result of .someval[index] this is never an attribute value,
-		                 // this is always a single config.
-		state_temporary, // the result of .length this value can never be written, it can only be read.
-
-	};
-
-	//Special case of std::enable_if
-	template<const variable_info_type vit>
-	struct enable_if_non_const
-	{
-		typedef enable_if_non_const<vit> type;
-	};
+public:
+	invalid_variablename_exception() : std::exception() {}
 
-	template<>
-	struct enable_if_non_const<vit_const>
+	const char* what() const NOEXCEPT
 	{
-	};
+		return "invalid_variablename_exception";
```

Comment:
iirc the point oif this line is just to to detect where the in the c++ code comes from in case some code forgot to catch this exception, this is never printed to the 'user'.